### PR TITLE
Add generic support for Device ID 0x0203 "Window Covering Controller" (e.g. Sunricher SR-ZG9080A)

### PR DESCRIPTION
**Pull request by slugzero**
This adds support for Sunricher window covering controller SR-ZG9080A cf.

https://sunricher.en.alibaba.com/product/62134471426-811954196/Zigbee_3_0_Zigbee_Curtain_Motor_Controller_Zigbee_Curtain_Controller_Smarthome.html
https://en.robbshop.nl/rollershutter-control-400w-zigbee-robb-smart?sqr=rollershutter%20controller&

Unbenannt2c

Unbenannt2a

Unbenannt2b

### CODE REVIEW

File: de_web_plugin.cpp [hunk 1]
```diff
@@ -218,6 +219,7 @@ static const SupportedDevice supportedDevices[] = {
     { VENDOR_SUNRICHER, "ICZB-RM", silabs2MacPrefix }, // iCasa remote
     { VENDOR_SUNRICHER, "ZGRC-KEY", emberMacPrefix }, // Sunricher wireless CCT remote
     { VENDOR_SUNRICHER, "ZG2833K", emberMacPrefix }, // Sunricher remote controller
+    { VENDOR_SUNRICHER, "Motor Controller", sunricherMacPrefix }, // Sunricher window covering controller
```
Comment (ebaauw) [inline de_web_plugin.cpp]: Is this needed?  The controller doesn't seem to have any cluster that's exposed as a `/sensors` resource.

**Comment by ebaauw** [comment]
Did you check the open/close logic?  Some devices have it reversed.  When fully closed, `state.bri` should be 0 and `state.on` should be `false`; when fully opened, `state.bri` should be 254.

File: de_web_plugin.cpp (hunk 1 above)
Comment (slugzero) [reply de_web_plugin.cpp]: You are right, this is not needed for the On/Off cluster. I was not sure about the others, though.

**Comment by mahaag** [comment]
I'm very interested in this device, but unfortunately for me the patch doesn't work. See the debug output of deConz (``deCONZ --dbg-info=1 --dbg-aps=1 --dbg-zcl=1 --dbg-zdp=1 --http-port=8080``) here: [deConz.txt](https://github.com/dresden-elektronik/deconz-rest-plugin/files/4208801/deConz.txt)

**Comment by manup** [comment]
Any news here?

**Comment by slugzero** [comment]
> Did you check the open/close logic? Some devices have it reversed. When fully closed, `state.bri` should be 0 and `state.on` should be `false`; when fully opened, `state.bri` should be 254.

If it is wired according to the manual (i"up" is connected to Q1 connector and "down" to Q2), it follows the proper logic. I.e. state.on is true when opened and false when closed.

**Comment by cgiraldo** [comment]
Hi! thanks a lot @slugzero, It works like a charm!

The only think to point out is that I need to use `{ "bri_inc":0, "on": true }` instead of `{"bri_inc":0}` to stop the cover while it is closing. 

If not the API returns an error with description `parameter, /lights/14/bri, is not modifiable. Device is set to off.`

By the way, @ebaauw, can you confirm this is the expected cover behavior for the deconz API?
 

> When fully closed, state.bri should be 0 and state.on should be false; when fully opened, state.bri should be 254. 

Because [deconz integration in home-assistant](https://github.com/home-assistant/core/blob/e696c08db05fd40789c495a70c3c35d4ab81313a/homeassistant/components/deconz/cover.py#L100) do it the other way around. If it is a home-assistant bug, I can fill an issue and provide a PR to fix it.

**Comment by Smanar** [comment]
@cgiraldo I m doing same ATM in PR list for profalux shutter
https://github.com/dresden-elektronik/deconz-rest-plugin/pull/2706/commits/6b57625dd10de353238b124eb34045e79ce0134e

**Comment by ebaauw** [comment]
> By the way, @ebaauw, can you confirm this is the expected cover behavior for the deconz API?

I'm afraid it's not, sorry for that.  Undated my comment.  This always confuses the hell out of me (I sure hope I get it right this time:
- In ZigBee, _Current Position Lift Percentage_ is supposed to be 0 for fully open and 100 for fully closed.  You could read it as "percentage closed".  However, Xiaomi and some other vendors have reversed this (using 0 for fully closed and 100 for fully open);
- The REST API should normalise these differences and expose `state.bri` as 0 for fully open;  `state.bri` as 254 for fully closed.  You could read it as "level closed".  As there's no Zigbee equivalent, `state.on` is derived from `state.bri` > 0, meaning: "not fully open";
- In HomeKit, _Target Position_ and _Current Position_ are 0 for closed and 100 for open.  You could read them as "percentage open".  Home lights up the tile when _Current Position_ > 0, so when "not fully closed";
- My latest PR adds `state.open` and `state.lift` to the `/lights` resource: `state.lift` follows the ZigBee attribute: 0 for open and 100 for closed (so "% closed").  `state.open` is derived from `state.lift` < 100 (so "not fully closed").  Note the subtlety here: `state.open` is **not** the inverse of `state.on`.
- Note that `state.on` and `state.bri` will still work, but should be considered deprecated.  I will only add support for `action.open` and `action.lift` to groups of window covering devices.

> Because deconz integration in home-assistant do it the other way around. If it is a home-assistant bug, I can fill an issue and provide a PR to fix it.

I don't know if HA uses "percentage open" or "percentage closed".  Best double-check; as I mentioned before, this always confuses the hell out of me.

**Comment by cgiraldo** [comment]
So with your updated comment, I can confirm the open/close logic is reversed for the Sunricher SR-ZG9080A. Should this issue be reopen or we create a new one just for SR-ZG9080A?

HA uses "percentage closed", so Its integration of the deconz REST API is correct.

**Pull request merged**